# HG changeset patch
# User Rik <rik@octave.org>
# Date 1466037265 25200
#      Wed Jun 15 17:34:25 2016 -0700
# Node ID b42b28d47f3180cc416950ec41ccac52af73a928
# Parent  c0604bba8da14e3400fea1e33063370595d5af31
# Parent  2c035963f469d16bd2e855e57739190e6065ea83
maint: Periodic merge of stable to default.

diff --git a/scripts/miscellaneous/ver.m b/scripts/miscellaneous/ver.m
--- a/scripts/miscellaneous/ver.m
+++ b/scripts/miscellaneous/ver.m
@@ -96,18 +96,18 @@ function retval = ver (package = "")
                                 "Release", [], "Date", lst{i}.date);
       endfor
     elseif (strcmpi (package, "Octave"))
       retval = struct ("Name", "Octave", "Version", version,
                        "Release", [], "Date", []);
     else
       lst = pkg ("list", package);
       if (isempty (lst))
-        retval = struct ("Name", "", "Version", [],
-                         "Release", [], "Date", []);
+        retval = struct ("Name", {}, "Version", {},
+                         "Release", {}, "Date", {});
       else
         retval = struct ("Name", lst{1}.name, "Version", lst{1}.version,
                          "Release", [], "Date", lst{1}.date);
       endif
     endif
   endif
 
 endfunction
@@ -127,8 +127,12 @@ endfunction
 %!   expected = lst{n}.name;
 %!   result = ver (expected);
 %!   assert (result.Name, expected);
 %!   assert (isfield (result, "Version"), true);
 %!   assert (isfield (result, "Release"), true);
 %!   assert (isfield (result, "Date"), true);
 %! endfor
 
+%!test
+%! result = ver ("%_an_unknown_package_%");
+%! assert (isempty (result), true);
+
